adafruit_mcp4728.py

```diff
@@ -281,59 +290,64 @@ class Channel:
 
     """
 
-    def __init__(self, dac_instance, cache_page, index):
+    def __init__(
+        self,
+        dac_instance: Literal[0, 1, 2, 3],
+        cache_page: Dict[str, int],
+        index: int,
+    ) -> None:
         self._vref = cache_page["vref"]
         self._gain = cache_page["gain"]
         self._raw_value = cache_page["value"]
         self._dac = dac_instance
         self.channel_index = index
 
     @property
-    def normalized_value(self):
+    def normalized_value(self) -> float:
         """The DAC value as a floating point number in the range 0.0 to 1.0."""
         return self.raw_value / (2**12 - 1)
 
     @normalized_value.setter
-    def normalized_value(self, value):
+    def normalized_value(self, value: float) -> None:
         if value < 0.0 or value > 1.0:
             raise AttributeError("`normalized_value` must be between 0.0 and 1.0")
 
         self.raw_value = int(value * 4095.0)
 
     @property
-    def value(self):
+    def value(self) -> int:
         """The 16-bit scaled current value for the channel. Note that the MCP4728 is a 12-bit piece
         so quantization errors will occur"""
         return self.normalized_value * (2**16 - 1)
 
     @value.setter
-    def value(self, value):
+    def value(self, value: int) -> None:
         if value < 0 or value > (2**16 - 1):
             raise AttributeError(
-                "`value` must be a 16-bit integer between 0 and %s" % (2**16 - 1)
+                f"`value` must be a 16-bit integer between 0 and {(2**16 - 1)}"
             )
 
         # Scale from 16-bit to 12-bit value (quantization errors will occur!).
         self.raw_value = value >> 4
 
     @property
-    def raw_value(self):
+    def raw_value(self) -> int:
         """The native 12-bit value used by the DAC"""
         return self._raw_value
 
     @raw_value.setter
-    def raw_value(self, value):
+    def raw_value(self, value: int) -> None:
         if value < 0 or value > (2**12 - 1):
             raise AttributeError(
-                "`raw_value` must be a 12-bit integer between 0 and %s" % (2**12 - 1)
+                f"`raw_value` must be a 12-bit integer between 0 and {(2**12 - 1)}"
             )
         self._raw_value = value
         # disabling the protected access warning here because making it public would be
         # more confusing
         self._dac._set_value(self)  # pylint:disable=protected-access
 
     @property
-    def gain(self):
+    def gain(self) -> Union[1, 2]:
```

Comment:
I think you meant to use `Literal` instead of `Union`, which is why this is failing the CI.